gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/DependencyRequirements.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.metric.schema.gradle;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.Multimap;
+import com.palantir.metric.schema.lang.LangMetricSchema;
+import com.palantir.metric.schema.lang.MetricSchemaCompiler;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+public final class DependencyRequirements {
+
+    public static Multimap<String, String> getDependencies(@Nonnull Collection<File> metricSchemaFiles) {
```

Comment:
nit: We generally assume values are not null, this feels unnecssary

```suggestion
    public static Multimap<String, String> getDependencies(Collection<File> metricSchemaFiles) {
```